app/src/main/java/com/google/android/sambadocumentsprovider/cache/DocumentCache.java

```diff
@@ -69,6 +69,15 @@ public void remove(Uri uri) {
     mCache.remove(uri);
     mExceptionCache.remove(uri);
 
+    // Do we need to delete it from its parent's children?
+    //
+    // There is situation where we delete a child from its parent's cache
+    // and the parent is queried while the cache is still valid. Consequently,
+    // we get the parent's sub folders without that child.
+    //
+    // Or may be we could just delete the parent's cache?
+    // mCache.remove(DocumentMetadata.buildParentUri(uri));
```

Comment:
I think either we delete its parent's cache, or leave it as is.

Effectively it's the same. Ejection shouldn't remove or delete anything, just remove a share from root list.